projects/igniteui-angular/src/lib/grids/row.directive.ts

```diff
@@ -58,6 +58,30 @@ export class IgxRowDirective<T extends IgxGridBaseDirective & GridType> implemen
     @Input()
     public index: number;
 
+    /**
+     * Gets whether the row is pinned.
+     * ```typescript
+     * let isPinned = row.pinned;
+     * ```
+     */
+    public get pinned(): boolean {
+        return this.grid.pinnedRecords.indexOf(this.rowData) !== -1;
+    }
+    /**
+     * Sets whether the row is pinned.
+     * Default value is `false`.
+     * ```typescript
+     * this.grid.selectedRows[0].pinned = true;
+     * ```
+     */
+    public set pinned(value: boolean) {
```

Comment:
Not sure this is needed, considering `row.grid` (and thus its API) is public + you have the shorthand pin methods.
Not a big deal though.

Reply:
I assume it was added in the specification so that it's the same as the column, although its functionality overlaps with the pin/unpin methods. @ChronosSF let me know if you still think it is necessary.

Reply:
I was specifically looking at the spec for this, but I can't see anything for the row API additions.

Reply:
While there's no dedicated section for Row API methods, there is a mention under 3.3. Developer Experience:

"Similar to the column counterparts, the row directive has pin and unpin methods that are used under the hood, as well as pinned getter/setter for maximum flexibility to the user. "

Reply:
Fair enough, still feels a bit redundant since the column is declarative and the row isn't, but one more shorthand won't hurt much.